conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java

```diff
@@ -182,24 +181,6 @@ public static List<Integer> newNonNullIntegerList(Iterable<Integer> iterable) {
         return integerList;
     }
 
-    // This method returns a list that can't handle nulls. Do not use this unless the nonNullCollections flag is set
-    public static List<Boolean> newNonNullBooleanList() {
-        return new ConjureBooleanList(new BooleanArrayList());
-    }
-
-    // This method returns a list that can't handle nulls. Do not use this unless the nonNullCollections flag is set
-    public static List<Boolean> newNonNullBooleanList(Iterable<Boolean> iterable) {
-        List<Boolean> booleanList;
-        if (iterable instanceof Collection) {
-            booleanList = new ConjureBooleanList(new BooleanArrayList(((Collection<Boolean>) iterable).size()));
-        } else {
-            booleanList = new ConjureBooleanList(new BooleanArrayList());
-        }
-        addAll(booleanList, iterable);
-
-        return booleanList;
-    }
```

Comment:
Removing these public apis is pretty dangerous, it will cause existing published code that has already been generated to fail at runtime when `conjure-lib` is upgraded.
They could instead be stubbed out as aliases to non-specialized `newNonNullList()`/etc.

Reply:
Hmm my thinking on this is that `conjure-lib` is bumped in lockstep with the code that generates the conjure classes. But thinking about that more deeply, even though they are housed in the same repo they are delivered by different mechanisms, `versions.props` vs a gradle plugin? So you are envisioning a scenario where `conjure-lib` is bumped by GCV, yet the code gen portions remain on an older gradle version, which has its own classpath resolution, and thus they can drift and this is dangerous?

Reply:
Not quite. Let’s say you publish an api, the types use this method. Your consumer which uses a client to your service upgrades all their conjure deps. 💥

Reply:
I addressed your concerns. I added a `@DoNotCall` I _think_ this be ignored for generated code which is the only legitimate use case. Correct me if I am wrong.